RubberduckTests/Grammar/VBAParserTests.cs

```diff
@@ -2859,6 +2859,21 @@ End Type
             AssertTree(parseResult.Item1, parseResult.Item2, "//commentOrAnnotation", matches => matches.Count == 1);
         }
 
+        [Category("Parser")]
+        [Test]
+        public void MidStatement()
```

Comment:
IMO, we need negative tests, too to assert that an expression that contains a `Mid()` function does not get parsed as a mid statement. We also need equivalent tests for `Mid$()` statement/function. 

For example, we definitely don't want something like `If Mid$("foo", 1) = "f" Then` to be treated as a `Mid` statement. The tests need to validate this.